### PR TITLE
/protocol organizational refactoring

### PR DESCRIPTION
**Pull request by c633**
Merge #170 & #182 first.

### CODE REVIEW

File: protocol/directory/directory.go [hunk 1]
```diff
@@ -6,14 +6,15 @@
...
-package protocol
+package directory
 
 import (
 	"bytes"
 
 	"github.com/coniks-sys/coniks-go/crypto/sign"
 	"github.com/coniks-sys/coniks-go/crypto/vrf"
 	"github.com/coniks-sys/coniks-go/merkletree"
+	p "github.com/coniks-sys/coniks-go/protocol"
```
Comment (masomel) [inline protocol/directory/directory.go]: Nit pick: We should probably be consistent with how we import some other coniks-go packages. We always import `protocol` as `p` which I think is fine, but `merkletree` sometimes is imported as `m` (e.g. https://github.com/coniks-sys/coniks-go/blob/master/protocol/consistencychecks.go#L13), and other times (like here) not. I think I've seen this with `directory` as well (`dir` vs the full name).

**Comment by masomel** [comment]
@c633 I went ahead and moved around the new test cases I added in #182 to make the tests pass again. There are, however, two tests in consistency_checks.go that no longer pass (`TestVerifyRegistrationWithTB` and `TestVerifyFulfilledPromise`). I don't know if you encountered this issue before.

File: protocol/directory/directory.go (hunk 1 above)
Comment (c633) [reply protocol/directory/directory.go]: Ok, I am fixing this.

File: protocol/client/consistencychecks.go [hunk 2]
```diff
@@ -2,15 +2,16 @@
 // on data received from a CONIKS directory.
 // These include data binding proof verification,
 // and non-equivocation checks.
-// TODO: move all STR-verifying functionality to a separate module
 
-package protocol
+package client
```
Comment (c633) [inline protocol/client/consistencychecks.go]: @masomel @Liamsi @arlolra I would like to have your inputs about this name. It is collided with our application client package (see https://github.com/coniks-sys/coniks-go/pull/183/files#diff-50a5a7c3748e6d839ee365d7f4291a1cR13). A name I think of is `verifier`. What do you think?

**Comment by c633** [comment]
Our error codes are a mess for now. Another PR will follow up.

**Comment by c633** [comment]
All tests are fixed. PTAL.

File: protocol/client/consistencychecks.go (hunk 2 above)
Comment (masomel) [reply protocol/client/consistencychecks.go]: Hmm. I think the auditor can also be considered a type of `verifier`, so I don't think this is the best name for this package. 

How about we change the client CLI package name? (Btw, we'll have to find a name other than `auditor` for the auditor CLI, too). Can we call it something like `client-bin` or `client-app`?

**Comment by c633** [comment]
> Are we going to go back to using import . for the test cases that used to have cyclical dependencies?

I think yes.

File: protocol/client/consistencychecks.go (hunk 2 above)
Comment (c633) [reply protocol/client/consistencychecks.go]: What do you think about https://github.com/coniks-sys/coniks-go/pull/183/commits/425366425260796134bdd9baef82e6fbc6933602?

**Comment by masomel** [comment]
In the spirit of refactoring: Is there an application layer API that we can refactor from the CLIs? I haven't spent much time looking into this, but it could be useful to have an application layer API for each component as well. This could then help developers integrate things like a client with their existing applications. What do you think?

**Comment by c633** [comment]
> Is there an application layer API that we can refactor from the CLIs?

Yes, it is. Actually, every module that doesn't belong to `cli` package should be used to provide APIs for developers. We do have plan for this, please see #101, I think it belongs to milestone 0.3 and 0.5.

**Pull request merged**